app.rb

```diff
@@ -0,0 +1,122 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def get_num(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def list_all_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+  end
+
+  def list_all_people
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 0.75
+  end
+
+  def create_person
+    print 'Do you want to create a student(1) or a teacher(2)? [Input the number]:'
+    input = gets.chomp.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+    case input
+    when 1
+      create_student(age, name)
+    when 2
+      create_teacher(age, name)
+    end
+    puts 'Person created successfully!'
+  end
+
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N]: '
+
+    parent_permission = gets.chomp.upcase == 'Y'
+    @people << Student.new(name, age, parent_permission)
+
+    puts 'Student created successfully'
+    sleep 0.75
+  end
+
+  def create_teacher(age, name)
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    @people << Teacher.new(specialization, age, name)
+
+    puts 'Teacher created successfully'
+    sleep 0.75
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+
+    puts 'Book added successfully'
+    sleep 0.75
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) #{book}" }
+
+    book_id = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index { |person, index| puts "#{index}) #{person}" }
```

Comment:
- [ ] The process of implementing the change requested initially you introduced an unexpected behavior when printing your books and person for creating rental. The prints in lines 96 and 101 print the **book** and **person** object memory reference, rather than the objects themselves (see image). This also prevents users from listing rental for a person. Kindly fix this.

  You may want to revert to what you previous had like so:
  ```
  @books.each_with_index { |book, index| puts "#{index}) #{book.title} #{book.author}" }
  
  @people.each_with_index { |person, index| puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
  ```
  ![somdo-rent-prints](https://user-images.githubusercontent.com/36542986/151939299-59748ee3-5ffd-4d8d-b58a-86dee52e9014.jpg)